.pre-commit-config.yaml

```diff
@@ -59,6 +59,13 @@ repos:
     - id: flake8
       args: ["--count", "--statistics"]
 
+-   repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.961
+    hooks:
+      - id: mypy
+        files: src
```

Comment:
In `pyproject.toml` you're adding

```
[tool.mypy]
files = "src"
```

so why do we need to specify the files here in `pre-commit`. It would be good to avoid duplicating this information to avoid potential future conflicts.

Reply:
This is not duplicating information (confusingly) and there's some awkward interplay between how pre-commit finds things and how mypy finds things. pre-commit is using `files` to determine when to call `mypy` and pass it in, but `mypy` ignores whatever `pre-commit` gives it (it seems) and just uses that. You're going to have to live with it.

Reply:
Lol. Amazing. Thanks for the summary though!